src/logging.cc

```diff
@@ -1381,15 +1383,28 @@ void LogMessage::Init(const char* file,
   //    (log level, GMT month, date, time, thread_id, file basename, line)
   // We exclude the thread_id for the default thread.
   if (FLAGS_log_prefix && (line != kNoLogPrefix)) {
-    stream() << LogSeverityNames[severity][0]
-             << setw(2) << 1+data_->tm_time_.tm_mon
-             << setw(2) << data_->tm_time_.tm_mday
-             << ' '
-             << setw(2) << data_->tm_time_.tm_hour  << ':'
-             << setw(2) << data_->tm_time_.tm_min   << ':'
-             << setw(2) << data_->tm_time_.tm_sec   << "."
-             << setw(6) << data_->usecs_
-             << ' '
+    stream() << LogSeverityNames[severity][0];
+    if (FLAGS_iso8601format) {
+      stream() << setw(2) << 1900+data_->tm_time_.tm_year << "-"
+               << setw(2) << 1+data_->tm_time_.tm_mon << "-"
+               << setw(2) << data_->tm_time_.tm_mday
+               << "T"
+               << setw(2) << data_->tm_time_.tm_hour << ':'
+               << setw(2) << data_->tm_time_.tm_min << ':'
+               << setw(2) << data_->tm_time_.tm_sec
+               << (data_->tm_time_.tm_gmtoff >= 0 ? "+" : "-")
+               << setw(2) << ::labs(data_->tm_time_.tm_gmtoff)/(60*60) << ":"
+               << setw(2) << ::labs(data_->tm_time_.tm_gmtoff)%(60*60);
```

Comment:
`tm_gmtoff` is not available for MSVC and MinGW.

Reply:
Is there any portable solution for `tm_gmtoff`?